src/test/record/test_frequency.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2021 Research Institute of Systems Planning, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from caret_analyze.record import ColumnValue
+from caret_analyze.record.frequency import Frequency
+from caret_analyze.record.record_factory import RecordFactory, RecordsFactory
+
+
+def create_records(records_raw, columns):
+    records = RecordsFactory.create_instance()
+    for column in columns:
+        records.append_column(column, [])
+
+    for record_raw in records_raw:
+        record = RecordFactory.create_instance(record_raw)
+        records.append(record)
+    return records
+
+
+def to_dict(records):
+    return [record.data for record in records]
+
+
+class TestFrequencyRecords:
+
+    def test_empty_case(self):
+        records_raw = [
+        ]
+        columns = [ColumnValue('timestamp')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+        ]
+        result = to_dict(frequency.to_records())
+        assert result == expect_raw
+
+    def test_one_column_default_case(self):
+        records_raw = [
+            {'timestamp': 0},
+            {'timestamp': 1},
+            {'timestamp': 11},
+            {'timestamp': 12},
+            {'timestamp': 14},
+        ]
+        columns = [ColumnValue('timestamp')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp': 0, 'frequency': 2},
+            {'timestamp': 10, 'frequency': 3}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=10))
+        assert result == expect_raw
+
+    def test_specify_interval_ns_case(self):
+        records_raw = [
+            {'timestamp': 1000000000},
+            {'timestamp': 1000500000},
+            {'timestamp': 2000000000},
+            {'timestamp': 2000300000},
+            {'timestamp': 3000000000},
+        ]
+        columns = [ColumnValue('timestamp')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp': 1000000000, 'frequency': 4},
+            {'timestamp': 3000000000, 'frequency': 1}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=2000000000))
```

Comment:
If my understanding is collecct, It's looks to me  that expected results are as follows.

```
        expect_raw = [
            {'timestamp': 1000000000, 'frequency': 2},
            {'timestamp': 3000000000, 'frequency': 3}
        ]
```

Are there any missing?

Reply:
I think the current output is correct.
Since `base_timestamp` is 1000000000 and `interval_ns` is 2000000000, the `'timestamp': 1000000000` row stores the number of timestamps in the range [1000000000, 3000000000).
Similarly, the `'timestamp': 3000000000` row stores the number of timestamps in the range [3000000000, 5000000000).

I have improved readability, could you please review it again?

Reply:
I see. LGTM!